lib/ui/widgets/web_link_card.dart

```diff
@@ -1,27 +1,32 @@
 // FLUTTER / DART / THIRD-PARTIES
 import 'package:flutter/material.dart';
-import 'package:flutter_inappwebview/flutter_inappwebview.dart';
 
-// CONSTANT
-import 'package:notredame/core/constants/router_paths.dart';
+// VIEWMODEL
+import 'package:notredame/core/viewmodels/web_link_card_viewmodel.dart';
 
 // MODEL
 import 'package:notredame/core/models/quick_link.dart';
 
-// SERVICE
-import 'package:notredame/core/services/navigation_service.dart';
-
 // OTHER
-import 'package:notredame/locator.dart';
 import 'package:notredame/ui/utils/app_theme.dart';
 
-class WebLinkCard extends StatelessWidget {
+class WebLinkCard extends StatefulWidget {
   final QuickLink _links;
 
-  /// used to redirect on the security.
-  final NavigationService _navigationService = locator<NavigationService>();
+  const WebLinkCard(this._links);
+
+  @override
+  _WebLinkCardState createState() => _WebLinkCardState();
+}
 
-  WebLinkCard(this._links);
+class _WebLinkCardState extends State<WebLinkCard> {
+  WebLinkCardViewModel model;
+
+  @override
+  void initState() {
+    super.initState();
+    model = WebLinkCardViewModel();
```

Comment:
Why do you create the view model here?
Why aren't you using a `ViewModelBuilder`?

Reply:
Bien vue ! :D